build-tools/docker/example-mainnet/docker-compose.yml

```diff
@@ -0,0 +1,101 @@
+version: '3'
+
+services:
+  node-daemon:
+    image: $DOCKERHUB_USERNAME/node-daemon:$ML_SOFTWARE_VERSION
+    volumes:
+      - ./container_home:/home/mintlayer
+    command: node-daemon mainnet
+    environment:
+      # Note:
+      # The default rpc bind address is '127.0.0.1', so it'll only be reachable from the same
+      # container; to make it reachable from other containers, we need to set it to '0.0.0.0'.
+      # For p2p, the default address is already '0.0.0.0', so this is not strictly needed; we
+      # do it for consistency and in order to specify the port explicitly (for extra readability).
+      # The latter has the additional benefit that it'll be easier to adapt this 'example-mainnet'
+      # for the testnet, which uses different ports by default.
+      - ML_MAINNET_NODE_RPC_BIND_ADDRESS=0.0.0.0:3030
+      - ML_MAINNET_NODE_P2P_BIND_ADDRESSES=0.0.0.0:3031
+      # wallet-cli can be run from this container as well, so we specify the rpc port for it
+      # explicitly too.
+      - ML_WALLET_NODE_RPC_ADDRESS=localhost:3030
+      - RUST_LOG
+    ports:
+      # Note: these are only needed if you want to access the node from the host system.
+      - "$NODE_RPC_HOST_PORT:3030"
+      - "$NODE_P2P_HOST_PORT:3031"
+
+  wallet-rpc-daemon:
+    image: $DOCKERHUB_USERNAME/wallet-rpc-daemon:$ML_SOFTWARE_VERSION
+    volumes:
+      - ./container_home:/home/mintlayer
+    command: wallet-rpc-daemon
+    depends_on:
+      - node-daemon
+    environment:
+      - ML_WALLET_RPC_DAEMON_NODE_RPC_ADDRESS=node-daemon:3030
+      # TODO: should wallet-rpc-daemon use the default cookie path automatically, like wallet-cli does?
+      - ML_WALLET_RPC_DAEMON_NODE_COOKIE_FILE=/home/mintlayer/.mintlayer/mainnet/.cookie
```

Comment:
So, `wallet-rpc-daemon` behaves differently from `wallet-cli` with respect to how the rpc-related parameters are handled. In `wallet-cli` if neither username/password nor cookie file is specified, a cookie file at the default location is assumed. But in `wallet-rpc-daemon` the absence of the parameters means "no authentication". Which is kind of strange, because `wallet-rpc-daemon` has a dedicated parameter for this (rpc_no_authentication). Should I go ahead and make `wallet-rpc-daemon` behave like  `wallet-cli` instead?

Reply:
I would prefer the default to be the user setting a default username and password. Put a comment above it saying something like "WARNING: This must be a strong password as it gives access to the wallet", or something like that.

Reply:
> I would prefer the default to be the user setting a default username and password. Put a comment above it saying something like "WARNING: This must be a strong password as it gives access to the wallet", or something like that.

I've changed the auth method for node rpc from the cookie to username/password. Regarding the comment - I'm not sure it's needed, because the default passwords ('use-strong-password', 'use-another-strong-password', 'use-yet-another-strong-password') already suggest that.